conversion/conversion.go

```diff
@@ -0,0 +1,789 @@
+	}
+}
+
+// getSeekable returns a seekable file (with same content as f) and the size of the content (in bytes).
+func getSeekable(f *os.File) (*os.File, int64, error) {
+	_, err := f.Seek(0, 0)
+	if err == nil { // Stdin is seekable, let's just use that. This happens when you run 'cmd < file'.
+		n, err := getSize(f)
+		return f, n, err
+	}
+	internal.VerbosePrintln("Creating a tmp file with a copy of stdin because stdin is not seekable.")
+
+	// Create file in os.TempDir. Its not clear this is a good idea e.g. if the
+	// pg_dump/mysqldump output is large (tens of GBs) and os.TempDir points to a directory
+	// (such as /tmp) that's configured with a small amount of disk space.
+	// To workaround such limits on Unix, set $TMPDIR to a directory with lots
+	// of disk space.
+	fcopy, err := ioutil.TempFile("", "harbourbridge.data")
+	if err != nil {
+		return nil, 0, err
+	}
+	syscall.Unlink(fcopy.Name()) // File will be deleted when this process exits.
+	_, err = io.Copy(fcopy, f)
+	if err != nil {
+		return nil, 0, fmt.Errorf("can't write stdin to tmp file: %w", err)
+	}
+	_, err = fcopy.Seek(0, 0)
+	if err != nil {
+		return nil, 0, fmt.Errorf("can't reset file offset: %w", err)
+	}
+	n, err := getSize(fcopy)
+	return fcopy, n, nil
+}
+
+// createDatabase returns a newly create Spanner DB.
+// It automatically determines an appropriate project, selects a
+// Spanner instance to use, generates a new Spanner DB name,
+// and call into the Spanner admin interface to create the new DB.
+func CreateDatabase(project, instance, dbName string, conv *internal.Conv, out *os.File) (string, error) {
+	fmt.Fprintf(out, "Creating new database %s in instance %s with default permissions ... ", dbName, instance)
+	ctx := context.Background()
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return "", fmt.Errorf("can't create admin client: %w", analyzeError(err, project, instance))
+	}
+	defer adminClient.Close()
+	// The schema we send to Spanner excludes comments (since Cloud
+	// Spanner DDL doesn't accept them), and protects table and col names
+	// using backticks (to avoid any issues with Spanner reserved words).
+	schema := conv.GetDDL(ddl.Config{Comments: false, ProtectIds: true})
+	op, err := adminClient.CreateDatabase(ctx, &adminpb.CreateDatabaseRequest{
+		Parent:          fmt.Sprintf("projects/%s/instances/%s", project, instance),
+		CreateStatement: "CREATE DATABASE `" + dbName + "`",
+		ExtraStatements: schema,
+	})
+	if err != nil {
+		return "", fmt.Errorf("can't build CreateDatabaseRequest: %w", analyzeError(err, project, instance))
+	}
+	if _, err := op.Wait(ctx); err != nil {
+		return "", fmt.Errorf("createDatabase call failed: %w", analyzeError(err, project, instance))
+	}
+	fmt.Fprintf(out, "done.\n")
+	return fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, dbName), nil
+}
+
+// getProject returns the cloud project we should use for accessing Spanner.
```

Comment:
Should be GetProject to match case of function definition.

Reply:
updated comment